acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/dsl/signer/EthSignerProcessRunner.java

```diff
@@ -187,4 +243,35 @@ private File createJsonFile(final String tempNamePrefix, final byte[] data) {
     keyFile.deleteOnExit();
     return keyFile;
   }
+
+  private void loadPortsFile() {
+    final File portsFile = new File(dataDirectory.toFile(), PORTS_FILENAME);
+    LOG.info("Awaiting presence of ethsigner.ports file: {}", portsFile.getAbsolutePath());
+    awaitPortsFile(dataDirectory);
+    LOG.info("Found ethsigner.ports file: {}", portsFile.getAbsolutePath());
+
+    awaitPortsFile(dataDirectory);
+
+    try (final FileInputStream fis = new FileInputStream(portsFile)) {
+      portsProperties.load(fis);
+      LOG.info("EthSigner ports: {}", portsProperties);
+    } catch (final IOException e) {
+      throw new RuntimeException("Error reading Pantheon ports file", e);
+    }
+  }
+
+  private void awaitPortsFile(final Path dataDir) {
+    final File file = new File(dataDir.toFile(), PORTS_FILENAME);
+    Awaitility.waitAtMost(30, TimeUnit.SECONDS)
+        .until(
+            () -> {
+              if (file.exists()) {
+                try (final Stream<String> s = Files.lines(file.toPath())) {
+                  return s.count() > 0;
+                }
+              } else {
```

Comment:
you can probably ditch the "else" and just return False

Reply:
done